### PR TITLE
windows: simplify vcbuild.bat

### PR DESCRIPTION
**Pull request by piscisaureus**
This patch is not ready for landing. It hasn't been fully polished!

cc @sblom for testing

### CODE REVIEW

**Comment by bnoordhuis** [comment]
Observations: doesn't apply against master and the commit log could use some love.

File: vcbuild.bat [hunk 1]
```diff
@@ -91,27 +65,18 @@ exit /b 1
...
+:build
+%PYTHON% gyp_uv -Dtarget_arch=%target_arch% -Dlibrary=%library% --build %config%
 goto run
 
-@rem Build the sln with msbuild.
-:msbuild-found
-msbuild uv.sln /t:%target% /p:Configuration=%config% /p:Platform="%platform%" /clp:NoSummary;NoItemAndPropertyList;Verbosity=minimal /nologo
-if errorlevel 1 exit /b 1
+:nobuild
+%PYTHON% gyp_uv -Dtarget_arch=%target_arch% -Dlibrary=%library% --build %config%
```
Comment (tjfontaine) [inline vcbuild.bat:78]: nobuild should still include --build?

**Comment by piscisaureus** [comment]
@sblom Did you manage to figure out what about this change broke the build on your machine?

**Comment by schlamar** [comment]
Does not work with VS 2010 Express:

```
>set GYP_MSVS_VERSION=2010e

>vcbuild.bat x86 release
['-Dtarget_arch=ia32', '-Dlibrary=static_library', '--build', 'Release', '.\\uv.gyp', '-I', '.\\common.gypi', '--depth=.', '-Dhost_arch=AMD64', '-Dcom
ponent=static_library']
Building [Release]: ['C:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\Common7\\IDE\\devenv.com', 'uv.sln', '/Build', 'Release']
Traceback (most recent call last):
  File "gyp_uv", line 100, in <module>
    run_gyp(gyp_args)
  File "gyp_uv", line 42, in run_gyp
    rc = gyp.main(args)
  File ".\build\gyp\pylib\gyp\__init__.py", line 527, in main
    return gyp_main(args)
  File ".\build\gyp\pylib\gyp\__init__.py", line 519, in gyp_main
    generator.PerformBuild(data, options.configs, params)
  File ".\build\gyp\pylib\gyp\generator\msvs.py", line 1835, in PerformBuild
    rtn = subprocess.check_call(arguments)
  File "C:\Python27\lib\subprocess.py", line 537, in check_call
    retcode = call(*popenargs, **kwargs)
  File "C:\Python27\lib\subprocess.py", line 524, in call
    return Popen(*popenargs, **kwargs).wait()
  File "C:\Python27\lib\subprocess.py", line 711, in __init__
    errread, errwrite)
  File "C:\Python27\lib\subprocess.py", line 948, in _execute_child
    startupinfo)
WindowsError: [Error 2] Das System kann die angegebene Datei nicht finden
```

**Comment by schlamar** [comment]
Reported upstream: https://code.google.com/p/gyp/issues/detail?id=387

While this is an issue I would just expect that "msbuild" is on PATH as in https://github.com/joyent/node/pull/6504.

File: vcbuild.bat [hunk 2]
```diff
@@ -91,27 +65,18 @@ exit /b 1
...
-@rem Check if VS build env is available
-if not defined VCINSTALLDIR goto msbuild-not-found
-goto msbuild-found
+if "%nobuild%"=="1" goto nobuild
+goto build
 
-:msbuild-not-found
-echo Build skipped. To build, this file needs to run from VS cmd prompt.
+:build
+%PYTHON% gyp_uv -Dtarget_arch=%target_arch% -Dlibrary=%library% --build %config%
```
Comment (rioki) [inline vcbuild.bat:74]: When building against head, it must be `gyp_uv.py` or else python will not find anything while forking. (At least for me.)

**Comment by rioki** [comment]
I just ran this `vcbuild.bat` against the head revision of master. (4ed6496b9c) I needed to change `gyp_uv` to `gyp_uv.py` and then it "worked".

I have VS2010 and VS2013 installed. (also VC6 ^_^) The build picked VS2010, which is not what I wanted. How does this script/gyp address the issue with multiple visual studios installed?

**Comment by saghul** [comment]
> I just ran this vcbuild.bat against the head revision of master. (4ed6496) I needed to change gyp_uv to gyp_uv.py and then it "worked".

That's because this PR is slightly behind master.

> I have VS2010 and VS2013 installed. (also VC6 ^_^) The build picked VS2010, which is not what I wanted. How does this script/gyp address the issue with multiple visual studios installed?

You can use the GYP_MSVS_VERSION env variable and set it to 2013 (or 2013e if you are running express version, I guess)

**Comment by saghul** [comment]
Based on the time of the last comment, I'm going to assume we can close this. Feel free to reopen if you continue working on this!

**Pull request closed without merging**